src/ai/backend/manager/models/endpoint.py

```diff
@@ -631,6 +646,107 @@ class Meta:
     items = graphene.List(Endpoint, required=True)
 
 
+class ModifyEndpointInput(graphene.InputObjectType):
+    resource_slots = graphene.JSONString()
+    resource_opts = graphene.JSONString()
+    cluster_mode = graphene.String()
+    cluster_size = graphene.Int()
+    desired_session_count = graphene.Int()
+    image = ImageRefType()
+    name = graphene.String()
+    resource_group = graphene.String()
+    open_to_public = graphene.Boolean()
+
+
+class ModifyEndpoint(graphene.Mutation):
+    class Arguments:
+        endpoint_id = graphene.UUID(required=True)
+        props = ModifyEndpointInput(required=True)
+
+    ok = graphene.Boolean()
+    msg = graphene.String()
+    endpoint = graphene.Field(lambda: Endpoint, required=False, description="Added since 23.09.9.")
+
+    @classmethod
+    async def mutate(
+        cls,
+        root,
+        info: graphene.ResolveInfo,
+        endpoint_id: uuid.UUID,
+        props: ModifyEndpointInput,
+    ) -> "ModifyEndpoint":
+        graph_ctx: GraphQueryContext = info.context
+        # raise ValueError(f"{type(graph_ctx.user) = }, {graph_ctx.user = }")
+        data: dict[str, Any] = {}
+        set_if_set(
+            props,
+            data,
+            "resource_slots",
+            clean_func=lambda v: ResourceSlot.from_user_input(v, None),
+        )
+        set_if_set(props, data, "resource_opts")
+        set_if_set(props, data, "cluster_mode")
+        set_if_set(props, data, "cluster_size")
+        set_if_set(props, data, "desired_session_count")
+        set_if_set(props, data, "image")
+        set_if_set(props, data, "resource_group")
+        image = data.pop("image", None)
+
+        async with graph_ctx.db.begin_readonly_session() as db_session:
+            if graph_ctx.user["role"] not in (UserRole.ADMIN, UserRole.SUPERADMIN):
+                user_id = graph_ctx.user["uuid"]
+                stmt = (
+                    sa.select(EndpointRow)
+                    .where(EndpointRow.id == endpoint_id)
+                    .where(
+                        (EndpointRow.session_owner == user_id)
+                        | (EndpointRow.created_user == user_id)
+                    )
+                )
+                endpoint_row = (await db_session.scalars(stmt)).first()
+                if endpoint_row is None:
+                    raise EndpointNotFound
+
```

Comment:
@fregataa Nice work, but still a little bit short to reach our goal - `ADMIN`s without `SUPERADMIN` privileges should not be able to alter configuration of endpoints from domains the account does not take control of.